src/main/java/org/veupathdb/eda/binaryfiles/dumper/multi/Main.java

```diff
@@ -38,17 +41,40 @@ public static void main(String[] args) throws Exception {
       throw new IllegalArgumentException(studiesDirectory.toAbsolutePath() + " is not a writable directory.");
     }
 
+    List<FailedStudy> failedStudies = new ArrayList<>();
+
     // instantiate a connection to the database
     try (DatabaseInstance appDb = new DatabaseInstance(SimpleDbConfig.create(
         SupportedPlatform.ORACLE, connectionUrl, connectionUser, connectionPassword))) {
 
       DataSource ds = appDb.getDataSource();
-      StudyFactory studyFactory = new StudyFactory(ds, APP_DB_SCHEMA, StudyOverview.StudySourceType.CURATED, false);
+      StudyFactory studyFactory = new StudyFactory(ds, APP_DB_SCHEMA, StudyOverview.StudySourceType.CURATED);
       for (StudyOverview studyOverview: studyFactory.getStudyOverviews()) {
-        Study study = studyFactory.getStudyById(studyOverview.getStudyId());
-        StudyDumper studyDumper = new StudyDumper(ds, APP_DB_SCHEMA, studiesDirectory, study);
-        studyDumper.dumpStudy();
+        try {
+          Study study = studyFactory.getStudyById(studyOverview.getStudyId());
+          StudyDumper studyDumper = new StudyDumper(ds, APP_DB_SCHEMA, studiesDirectory, study);
+          studyDumper.dumpStudy();
+        } catch (Exception e) {
+          failedStudies.add(new FailedStudy(e, studyOverview));
+        }
       }
     }
+    if (!failedStudies.isEmpty()) {
+      final String commaSeparatedStudies = failedStudies.stream()
+          .map(study -> study.studyOverview.getStudyId())
+          .collect(Collectors.joining(","));
+      throw new RuntimeException("The following studies failed to be written: " + commaSeparatedStudies
+          + ". Attaching first failure stack trace: ", failedStudies.stream().findFirst().get().e);
+    }
+  }
+
+  public static class FailedStudy {
```

Comment:
I often extend TwoTuple in cases like this to save a few lines.  Members are available via getFirst/getKey and getSecond/getValue.  Not sure if it's a bad habit though. e.g.
```
private static class FailedStudy extends TwoTuple<Exception, StudyOverview> {
    public FailedStudy(Exception first, StudyOverview second) { super(first, second); }
}

Reply:
I might slightly prefer to keep as is to make the getters more clear. I'm open to update if you feel strongly about it though